pkg/agent/upgrader/kubernetes/containers.go

```diff
@@ -0,0 +1,81 @@
+package kubernetes
+
+import (
+	"context"
+
+	"golang.org/x/exp/slices"
+	"google.golang.org/protobuf/types/known/emptypb"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (k *kubernetesAgentUpgrader) getAgentDeployment(ctx context.Context) (*appsv1.Deployment, error) {
+	deployment := &appsv1.Deployment{}
+	err := k.k8sClient.Get(ctx, client.ObjectKey{
+		Name:      "opni-agent",
+		Namespace: k.namespace,
+	}, deployment)
+	if err != nil {
+		return nil, err
+	}
+	return deployment, nil
+}
+
+func (k *kubernetesAgentUpgrader) getAgentContainer(containers []corev1.Container) *corev1.Container {
+	for _, container := range containers {
+		if slices.Contains(container.Args, "agentv2") {
```

Comment:
We're explicitly checking the arguments rather than the container names as these are less arbitrary.  I made a conscious choice not to provide a full manifest and instead mutate an existing one as number of things are arbitrary now and are decided at install time.  It may turn out that we want to manage more of this in the future, but I'm against it initially.

Reply:
ah yeah that makes sense.

Maybe this logic on the agent-side should be put in its own agent plugin so that when the gateway broadcasts changes to plugins, the agent always runs the matching implementation of the upgrader service, regardless of the environment its running in

Reply:
this probably doesn't need to be addressed any time soon but it feels like the initial abstraction we stick with could be important

Reply:
Yeah the idea was to delegate all of the upgrade logic to the driver.  It can choose to access the image resolver or not; this is a generic endpoint.  I wanted to avoid a tight coupling between agent and gateway for the upgrader as I think this paints us into more of a corner and is more complicated.

Reply:
sounds good to me